src/Strongly.Options.SourceGenerators/Extensions/AssemblySymbolExtensions.cs

```diff
@@ -0,0 +1,33 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Strongly.Options.SourceGenerators.Extensions;
+
+using static WellKnownNamings;
+
+public static class AssemblySymbolExtensions
+{
+    public static string GetModuleNameFromAttributeOrShortAssemblyNameByDefault(this IAssemblySymbol assembly)
+    {
+        var userSpecifiedModuleName = assembly
+           .GetAttributes()
+           .FirstOrDefault(x => x.AttributeClass!.ToDisplayString() == StronglyOptionsModuleAttribute)?
+           .ConstructorArguments
+           .FirstOrDefault()
+           .Value?
+           .ToString();
+
+        return userSpecifiedModuleName ?? assembly.GetLastAssemblyNamePart();
```

Comment:
```suggestion
        return userSpecifiedModuleName ?? assembly.GetLastPartOfAssemblyName();
```
как будто Last и Part слишком далеко друг от друга находятся, когда до Part дочитываешь, то смысл, что он 'Last', уже забывается

Reply:
даже наверное `GetLastPartOfName`